src/quickgui/qgsquickutils.h

```diff
@@ -57,27 +60,68 @@ class QUICK_EXPORT QgsQuickUtils: public QObject
   public:
     //! Create new utilities
     QgsQuickUtils( QObject *parent = nullptr );
-    //! dtor
+    //! Destructor
     ~QgsQuickUtils() = default;
 
     //! \copydoc QgsQuickUtils::dp
     qreal screenDensity() const;
 
     /**
-      * Calculate the distance in meter representing baseLengthPixels pixels on the screen based on the current map settings.
+      * Creates crs from epsg code in QML
       */
-    Q_INVOKABLE double screenUnitsToMeters( QgsQuickMapSettings *mapSettings, int baseLengthPixels ) const;
+    Q_INVOKABLE static QgsCoordinateReferenceSystem coordinateReferenceSystemFromEpsgId( long epsg );
+
+    /**
+      * Creates QgsPointXY in QML
+      */
+    Q_INVOKABLE QgsPointXY pointXYFactory( double x, double y ) const;
+
+    /**
+      * Creates QgsPoint in QML
+      */
+    Q_INVOKABLE QgsPoint pointFactory( double x, double y ) const;
+
+    /**
+      * Converts QGeoCoordinate to QgsPoint
+      */
+    Q_INVOKABLE QgsPoint coordinateToPoint( const QGeoCoordinate &coor ) const;
+
+    /**
+      * Transforms point between different crs from QML
+      */
+    Q_INVOKABLE static QgsPointXY transformPoint( const QgsCoordinateReferenceSystem &srcCrs,
+        const QgsCoordinateReferenceSystem &destCrs,
+        const QgsCoordinateTransformContext &context,
+        const QgsPointXY &srcPoint );
+
+    /**
+      * Calculates the distance in meter representing baseLengthPixels pixels on the screen based on the current map settings.
+      */
+    Q_INVOKABLE static double screenUnitsToMeters( QgsQuickMapSettings *mapSettings, int baseLengthPixels );
 
     //! Log message in QgsMessageLog
     Q_INVOKABLE void logMessage( const QString &message,
                                  const QString &tag = QString( "QgsQuick" ),
                                  Qgis::MessageLevel level = Qgis::Warning );
 
     /**
-     * Returns a string with information about screen size and resolution
-     *
-     * Useful to log for debugging of graphical problems on various display sizes
-     */
+      * Returns QUrl to image from libary's /images folder.
+      */
+    Q_INVOKABLE QUrl getThemeIcon( const QString &name );
+
+    /**
+      * Converts point to string with given decimals (default decimals = 3),
+      * e.g. -2.234521, 34.4444421 -> -2.234, 34.444
+      */
+    Q_INVOKABLE static QString qgsPointToString( const QgsPoint &point, int decimals = 3 );
```

Comment:
probably yes, but for this phase we tried to not modify core libraries

Reply:
What's the reason for the "phase" argument?
I'd prefer to do things in the right spot from the beginning to avoid having to refactor code, it's a PITA to get rid of methods later on.

Is this a method that is of general interest for

 * any QgisQuick application
 * QGIS in general
 * the QgisQuick demo app

Reply:
Make sense, I will be fearless next time!

Looking at the usage it seems to me that it is of general interest for any QgsiQuick application to show GPS location to an user, but I am not sure if it has any great benefit for QGIS in general. So I kept it here.

Reply:
Ẅhat about QgsCoordinateFormatter (or an improved version of it)? CC @nyalldawson

Reply:
Good to know such class exists :)

Looking at https://qgis.org/api/qgscoordinateformatter_8cpp_source.html it looks like the implementation is the same. Are you proposing to use QgsCoordinateFormatter in qgsPointToString implementation instead of what is there? Or more like adding distanceToString to QgsCoordinateFormatter too and have it as new quick helper class?

Reply:
used QgsCoordinateFormatter in the last commit for transformation. I keep it in QgsQuickUtils so we do not need to create another singleton for those static functions